### PR TITLE
chore: Upgrade Airbnb ESLint config versions

### PR DESCRIPTION
**Pull request by taion**
BREAKING CHANGE: This probably breaks something.

### CODE REVIEW

File: packages/eslint-config-4catalyzer-react/package.json [hunk 1]
```diff
@@ -29,20 +29,21 @@
...
-    "eslint-config-airbnb": "^17.1.1",
-    "eslint-plugin-react-hooks": "^1.6.1"
+    "eslint-config-airbnb": "^18.0.1"
   },
   "peerDependencies": {
     "eslint-plugin-import": "^2.18.2",
     "eslint-plugin-jsx-a11y": "^6.2.3",
-    "eslint-plugin-react": "^7.14.3"
+    "eslint-plugin-react": "^7.14.3",
+    "eslint-plugin-react-hooks": "^1.7.0"
```
Comment (taion) [inline packages/eslint-config-4catalyzer-react/package.json]: should have been a peer dep all along

File: packages/eslint-config-4catalyzer-react/rules.js [hunk 2]
```diff
@@ -1,16 +1,6 @@
 module.exports = {
   // We put all styling imports at the end.
   'import/first': 'off',
-  // Allow <Link to>.
-  // TODO: Remove this when airbnb includes this in next release.
-  'jsx-a11y/anchor-is-valid': [
```
Comment (taion) [inline packages/eslint-config-4catalyzer-react/rules.js]: fixed upstream

File: packages/eslint-config-4catalyzer-react/rules.js [hunk 3]
```diff
@@ -21,9 +11,6 @@ module.exports = {
       depth: 25,
     },
   ],
-  'jsx-a11y/label-has-for': 'off',
```
Comment (taion) [inline packages/eslint-config-4catalyzer-react/rules.js]: fixed upstream

File: packages/eslint-config-4catalyzer-react/rules.js [hunk 4]
```diff
@@ -21,9 +11,6 @@ module.exports = {
...
-  'react-hooks/rules-of-hooks': 'error',
```
Comment (taion) [inline packages/eslint-config-4catalyzer-react/rules.js]: included in upstream

File: packages/eslint-config-4catalyzer-react/rules.js [hunk 5]
```diff
@@ -21,9 +11,6 @@ module.exports = {
...
-  'react-hooks/exhaustive-deps': 'warn',
```
Comment (taion) [inline packages/eslint-config-4catalyzer-react/rules.js]: this is an error upstream, but we don't allow errors in our pre-commit hooks mostly, so who cares

File: packages/eslint-config-4catalyzer-react/rules.js [hunk 6]
```diff
@@ -61,6 +50,7 @@ module.exports = {
         'getters',
         'setters',
         '/^(get|set)(?!(InitialState$|DefaultProps$|ChildContext$)).+$/',
+        'instance-methods',
```
Comment (taion) [inline packages/eslint-config-4catalyzer-react/rules.js]: why not

File: packages/eslint-config-4catalyzer/rules.js [hunk 7]
```diff
@@ -23,7 +21,6 @@ module.exports = {
     },
   ],
   'no-plusplus': 'off',
-  'no-restricted-globals': ['error'].concat(restirctedGlobals),
```
Comment (taion) [inline packages/eslint-config-4catalyzer/rules.js]: AFAICT this was just to allow `isNaN` and `isFinite`? But they both legitimately are sort of crappy.

**Comment by taion** [comment]
Did we explicitly decide against `array-type` and `no-this-alias` for TS?

File: packages/eslint-config-4catalyzer-react/rules.js (hunk 5 above)
Comment (jquense) [reply packages/eslint-config-4catalyzer-react/rules.js]: only frontend does the no warnings thing, which isn't tenable for telemed right now, it's got a lot of warning that are intentionally left there (e.g. alerts, console.logs)

File: packages/eslint-config-4catalyzer/rules.js (hunk 7 above)
Comment (jquense) [reply packages/eslint-config-4catalyzer/rules.js]: not sure i understand, this prevents you from using sketchy globals like `name` and `event` things that are likely local identifiers and you'd not get a "undef" warning for if you forgot to actually declare them in code

File: packages/eslint-config-4catalyzer-react/rules.js (hunk 4 above)
Comment (jquense) [reply packages/eslint-config-4catalyzer-react/rules.js]: nice

File: packages/eslint-config-4catalyzer-react/rules.js (hunk 6 above)
Comment (jquense) [reply packages/eslint-config-4catalyzer-react/rules.js]: does this make you need to group arrow functions and methods? That was one of the motivating reasons to override this.

File: packages/eslint-config-4catalyzer-react/rules.js (hunk 5 above)
Comment (taion) [reply packages/eslint-config-4catalyzer-react/rules.js]: okay, rather, it gets auto-fixed, so we still need to forcibly pragma it

File: packages/eslint-config-4catalyzer-react/rules.js (hunk 6 above)
Comment (taion) [reply packages/eslint-config-4catalyzer-react/rules.js]: i thought we overrode this to insert the thing for type definitions

File: packages/eslint-config-4catalyzer/rules.js (hunk 7 above)
Comment (taion) [reply packages/eslint-config-4catalyzer/rules.js]: no, i mean, this was already defined upstream: https://github.com/airbnb/javascript/blob/21b65e943c361f91af5a484460fb2c8d02e2ff92/packages/eslint-config-airbnb-base/rules/variables.js#L19

Comment (jquense) [reply packages/eslint-config-4catalyzer/rules.js]: o nice, 👍 didn't know that, just duplicated work then

**Comment by taion** [comment]
@jquense 	i changed the TS config to extend and override the base rule. let me know if this is gtg

File: packages/eslint-config-4catalyzer-react/rules.js (hunk 6 above)
Comment (jquense) [reply packages/eslint-config-4catalyzer-react/rules.js]: also that

**Comment by jquense** [comment]
lgtm if the parser setup is the same i'm guessing in the base configs ?

**Comment by taion** [comment]
yup: https://github.com/typescript-eslint/typescript-eslint/blob/ca3b6a57b589213c6761cc311127cf3aa7e42082/packages/eslint-plugin/src/configs/base.json#L2-L4

**Comment by taion** [comment]
i force reverted this because i don't want to major-bump everything. gah

**Pull request merged**